src/main/java/com/direwolf20/buildinggadgets/common/registry/OurItems.java

```diff
@@ -0,0 +1,132 @@
+
+    // Back to how the actual class works.
+    private static Set<Builder> itemBuilders = new HashSet<>();
+
+    /**
+     * setup us required to be loaded before the registry event
+     * happens.
+     */
+    public static void setup() {
+        // Looks complicated but it's not. We're just building a list of Builders that will, when
+        // applied will construct the Item.Properties Builder for us allowing us to do a lot less work.
+        itemBuilders.addAll(new HashSet<Builder>() {{
+            add(new Builder(ItemReference.GADGET_BUILDING_RL).setBuilder(nonStackableItemProperties().maxDamage(1)).setFactory(GadgetBuilding::new));
+            add(new Builder(ItemReference.GADGET_EXCHANGING_RL).setBuilder(nonStackableItemProperties().maxDamage(1)).setFactory(GadgetExchanger::new));
+            add(new Builder(ItemReference.GADGET_COPY_PASTE_RL).setBuilder(nonStackableItemProperties().maxDamage(1)).setFactory(GadgetCopyPaste::new));
+            add(new Builder(ItemReference.GADGET_DESTRUCTION_RL).setBuilder(nonStackableItemProperties().maxDamage(1)).setFactory(GadgetDestruction::new));
+            add(new Builder(ItemReference.PASTE_CONTAINER_T1_RL).setBuilder(nonStackableItemProperties()).setFactory(RegularPasteContainerTypes.T1::create));
+            add(new Builder(ItemReference.PASTE_CONTAINER_T2_RL).setBuilder(nonStackableItemProperties()).setFactory(RegularPasteContainerTypes.T2::create));
+            add(new Builder(ItemReference.PASTE_CONTAINER_T3_RL).setBuilder(nonStackableItemProperties()).setFactory(RegularPasteContainerTypes.T3::create));
+            add(new Builder(ItemReference.PASTE_CONTAINER_CREATIVE_RL).setBuilder(nonStackableItemProperties()).setFactory(ConstructionPasteContainerCreative::new));
+            add(new Builder(ItemReference.CONSTRUCTION_PASTE_RL).setBuilder(itemProperties()).setFactory(ConstructionPaste::new));
+            add(new Builder(ItemReference.CONSTRUCTION_CHUNK_DENSE_RL).setBuilder(itemProperties()).setFactory(Item::new));
+            add(new Builder(ItemReference.TEMPLATE_RL).setBuilder(itemProperties()).setFactory(Template::new));
+        }});
+    }
+
+    /**
+     * Register the items with the forge registry by looping over
+     * our builders and calling their register method.
+     *
+     * @param event the forge registry event
+     */
+    @SubscribeEvent
+    public static void registerItems(RegistryEvent.Register<Item> event) {
+        itemBuilders.forEach(e -> event.getRegistry().register(
+                e.register()
+        ));
+    }
+
+    static Item.Properties itemProperties() {
+        return new Item.Properties().group(RegistryHandler.creativeTab);
+    }
+
+    private static Item.Properties nonStackableItemProperties() {
+        return itemProperties().maxStackSize(1);
+    }
+
+    /**
+     * A very simple Builder for item registry to provide
+     * forge / mc with our items.
+     */
+    private static final class Builder {
```

Comment:
Could aou maybe make everything in that class private? Because the class is private you already cannot access it outside the file, but it's clearer in my opinion.

Reply:
Sure